redfish_client/resource.py

```diff
@@ -109,6 +117,45 @@ def execute_action(self, action_name, payload):
             return self._connector.post(action.target, payload=payload)
         raise KeyError("Action with {} does not exist".format(action_name))
 
+    def wait_for_stat(self, stat, expected_value, fail_values=list(), poll_every_num_sec=3,
```

Comment:
Using empty list as a default function parameter is not the safest thing to do. It is OK in our case, since we are only reading from it, but I would rather replace it with empty tuple that is immutable. Why? Try to run this:

    def f(a=[]):
        print(a)
        a.append("a")

    for _ in range(5):
        f()